builds/jenkins-build-sls-app/Jenkinsfile

```diff
@@ -0,0 +1,1062 @@
+      events.sendStartedEvent('DEPLOY_TO_AWS', 'Deployment started to staging AWS environment', context_map, environment_logical_id)
+      events.sendStartedEvent('UPDATE_ENVIRONMENT', "Environment status update event for ${env_key} deployment", environmentDeploymentMetadata.generateEnvironmentMap("deployment_started", environment_logical_id, null), environment_logical_id)
+
+      echo "starts Deployment to ${env_key} Env"
+      def lambdaARN = null
+      environmentDeploymentMetadata.setEnvironmentEndpoint(lambdaARN)
+
+      events.sendStartedEvent('UPDATE_ENVIRONMENT', "Environment status update event for ${env_key} deployment", environmentDeploymentMetadata.generateEnvironmentMap("deployment_started", environment_logical_id, null), environment_logical_id)
+
+      withCredentials([
+        [$class: 'AmazonWebServicesCredentialsBinding', accessKeyVariable: 'AWS_ACCESS_KEY_ID', credentialsId: configLoader.AWS_CREDENTIAL_ID, secretKeyVariable: 'AWS_SECRET_ACCESS_KEY']
+      ]) {
+        try {
+          // initialize aws credentials
+          def randomString = utilModule.generateRequestId();
+          def credsId = "jazz-${randomString}";
+          sh "aws configure set profile.${credsId}.region ${configLoader.AWS.REGION}"
+          sh "aws configure set profile.${credsId}.aws_access_key_id $AWS_ACCESS_KEY_ID"
+          sh "aws configure set profile.${credsId}.aws_secret_access_key $AWS_SECRET_ACCESS_KEY"
+
+          // Generate serverless yml file with domain added in function name
+          echo "Generate deployment env with domain"
+
+          if (isScheduleEnabled) {
+            def eventsArns = getEventsArn(config, environment_logical_id)
+            for (def i = 0; i < eventsArns.size(); i++) {
+              def arn = eventsArns[i]
+              events.sendCompletedEvent('CREATE_ASSET', null, utilModule.generateAssetMap("aws", arn, "cloudwatch_event", config), environment_logical_id);
+            }
+          }
+
+          def envBucketKey = "${env_key}${configLoader.JAZZ.S3_BUCKET_NAME_SUFFIX}"
+          echoServerlessFile()
+          def deployOutput = servelessDeploy(environment_logical_id, envBucketKey)
+
+// Shoving deploy output lines into array
+          def deployLines = deployOutput.split('\n')
+          def resourceArns = []
+          def afterStackOutputs = false
+// Extracting all the lines that follow 'Stack Outputs'. These lines are supposed to contain arns of resources created
+          for(def line: deployLines) {
+            if(afterStackOutputs) resourceArns.add(line)
+            if(line.contains('Stack Outputs')) afterStackOutputs = true // Here we are only interested in lines that follow 'Stack Outputs' marker
+          }
+
+          for(def name2ArnString: resourceArns) {
+            def name2ArnPair = name2ArnString.split(': ') // Here we rely on the space after the ': ' so it is split properly without affecting the arn itself that is full of colons but never has a space after
```

Comment:
Can we move this logic to a function that takes in ARN & returns a map of all the values  that we care about - $type, $region, $resourceName etc.?

Reply:
Well so far I simply needed 'type' only and so the 
```groovy
                 if(arnAsArray.size() > 2) { // Making sure that the third element exists
                   def artifactType = arnAsArray[2]
                 .......
                }
```
looks like a simplest way to achieve the goal. However if the need to parse the 'arn' arises again we can sure create a common function and put it somewhere inside 'jazz-build-module'